spec/features/hiring_staff_can_edit_a_draft_vacancy_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+RSpec.feature 'Hiring staff can edit a draft vacancy' do
+  let(:school) { create(:school) }
+  before do
+    stub_hiring_staff_auth(urn: school.urn)
+  end
+
+  context 'with job specification completed' do
+    let!(:vacancy) do
+      pay_scales = create_list(:pay_scale, 3)
+      VacancyPresenter.new(build(:vacancy, :complete,
+                                 job_title: 'Draft vacancy',
+                                 school: school,
+                                 min_pay_scale: pay_scales.sample,
+                                 max_pay_scale: pay_scales.sample,
+                                 working_patterns: ['full_time', 'part_time']))
+    end
+    let(:draft_vacancy) { Vacancy.find_by(job_title: vacancy.job_title) }
+
+    before do
+      visit new_school_job_path
+      fill_in_job_specification_form_fields(vacancy)
+      click_on I18n.t('buttons.save_and_continue')
+    end
+
+    scenario 'redirects to incomplete candidate specification step, with fields pre-populated' do
+      draft_vacancy.education = 'Teaching degree'
+      draft_vacancy.qualifications = 'New Teacher Qualification'
+      draft_vacancy.save(validate: false)
+
+      visit edit_school_job_path(id: draft_vacancy.id)
+
+      expect(page).to have_content('Step 2 of 3')
+      expect(page).to have_content(draft_vacancy.education)
+      expect(page).to have_content(draft_vacancy.qualifications)
+    end
+
+    context 'when editing a different vacancy' do
+      # We use the session to store vacancy attributes, make sure it doesn't leak between edits.
```

Comment:
I wonder if this comment could be described in the context instead, although its quite long 🤔